radix-transactions/src/manifest/static_resource_movements/visitor.rs

```diff
@@ -0,0 +1,724 @@
+    /// The buckets tracked by the by the visitor.
+    tracked_buckets: IndexMap<ManifestBucket, BucketContent>,
+    /// The information about the invocations observed in this manifest. This will be surfaced to
+    /// the user when they call the output function.
+    invocation_static_information: IndexMap<usize, InvocationStaticInformation>,
+}
+
+impl StaticResourceMovementsVisitor {
+    pub fn new(initial_worktop_state_is_unknown: bool) -> Self {
+        let worktop_uncertainty_sources = if initial_worktop_state_is_unknown {
+            vec![WorktopUncertaintySource::YieldFromParent]
+        } else {
+            vec![]
+        };
+        Self {
+            tracked_buckets: Default::default(),
+            worktop_fungible_contents: Default::default(),
+            worktop_non_fungible_contents: Default::default(),
+            worktop_uncertainty_sources,
+            invocation_static_information: Default::default(),
+        }
+    }
+
+    pub fn output(self) -> StaticResourceMovementsOutput {
+        StaticResourceMovementsOutput {
+            invocation_static_information: self.invocation_static_information,
+        }
+    }
+
+    fn handle_invocation(
+        &mut self,
+        kind: InvocationKind<'_>,
+        args: &ManifestValue,
+        index: usize,
+    ) -> ControlFlow<StaticResourceMovementsError<'static>, InvocationStaticInformation> {
+        // Creating a new invocation static information which will be returned back to the caller at
+        // the end of this handling.
+        let mut invocation_information = InvocationStaticInformation {
+            kind: kind.into(),
+            input: Default::default(),
+            output: Default::default(),
+        };
+
+        // Get the invocation inputs based on the arguments.
+        invocation_information.input = self.resolve_args_into_invocation_io(args)?;
+
+        // Creating a typed native invocation to use in interpreting the invocation.
+        let typed_native_invocation = match kind {
+            InvocationKind::Method {
+                address: DynamicGlobalAddress::Static(global_address),
+                module_id,
+                method,
+            } => TypedNativeInvocation::from_method_invocation(
+                global_address.as_node_id(),
+                module_id,
+                method,
+                args,
+            )
+            .map(|value| (value, global_address.into_node_id())),
+            InvocationKind::Function {
+                address: DynamicPackageAddress::Static(package_address),
+                blueprint,
+                function,
+            } => TypedNativeInvocation::from_function_invocation(
+                package_address.as_node_id(),
+                blueprint,
+                function,
+                args,
+            )
+            .map(|value| (value, package_address.into_node_id())),
+            // Can't convert into a typed native invocation.
```

Comment:
(Not done, can still be picked up)

Reply:
Very true, I think we should add this as a todo instead of implementing it in this PR due to address reservations not being used too much at the moment.

Reply:
Address in #1931